src/archemist/core/optimisation/recipe_converter.py

```diff
@@ -0,0 +1,96 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+import logging
+import re
+
+class RecipeUpdate:
+    def __init__(self, template_path, recipe_path) -> None:
+        self.recipe_dict  = YamlHandler.load_recipe_file(template_path)
+        self.recipe_dir = recipe_path
+        self.placeholders = []
+        self.placeholder_keys_all = []
+        self.placeholder_keys_any = []
+        self.find_placeholders(self.recipe_dict)
+
+    def find_placeholders(self, recipe_dict):
+        if isinstance(recipe_dict, dict):
+            for key, value in recipe_dict.items():
+                if isinstance(value, str) and re.search(r"^{{*.*}}$", value):
+                    _strings = re.split("\.", value)
+                    _data_type = _strings[0].lstrip("{")
+                    _out_type = _strings[1].rstrip("}")
+                    if _out_type == "all":
+                        self.placeholder_keys_all.append(key)
+                        self.placeholders.append(value)
+                    elif _out_type == "any":
+                        self.placeholder_keys_any.append(key)
+                        self.placeholders.append(value)
+                else:
+                    self.find_placeholders(value)
+        elif isinstance(recipe_dict, list):
+            for item in recipe_dict:
+                self.find_placeholders(item)
+    
+    def generate_recipe(self, _pd):
```

Comment:
can you please more descriptive name instead of "_pd" and use type hint as suggested earlier.

Reply:
changed to optimised_parameters_data